src/tests/ftest/process_cores.py

```diff
@@ -0,0 +1,454 @@
+
+        """
+        info = []
+        for item in (self.name, self.version, self.release):
+            if item is None:
+                break
+            info.append(item)
+        return "-".join(info)
+
+
+def get_core_files(args):
+    """Get the names of the core files on the remote hosts and their total 1k block size.
+
+    Args:
+        args (argparse.Namespace): command line arguments for this program
+
+    Raises:
+        RuntimeError: if there is a problem listing core.* files on the remote hosts
+
+    Returns:
+        tuple: the total 1k block size required for all of the core files, a list of lists of core
+            files detected per set of nodes
+
+    """
+    required_1k_blocks = 0
+    core_files = []
+
+    # Get a list of any core files and their size in 1k blocks
+    command = ["ls", "-1sk", os.path.join(args.source, "core.*")]
+    task = get_remote_output(list(args.hosts), " ".join(command))
+    if not check_remote_output(task, command):
+        raise RuntimeError("Error detecting core files on {}".format(args.hosts))
+
+    # Populate a dictionary of active interfaces with a NodSet of hosts on which it was found
+    for output, nodelist in task.iter_buffers():
+        # nodeset = NodeSet.fromlist(nodelist)
+        core_files.append([NodeSet.fromlist(nodelist)])
+        display(args, "Core files detected on {}:".format(str(core_files[-1][0])), 0)
+        for line in output:
+            match = re.findall(r"\s+(\d+)\s+(core.*)", line.decode("utf-8"))
+            if not match:
+                display(args, "  - No core files found", 0)
+                continue
+            for item in match:
+                try:
+                    size = int(item[0])
+                except (ValueError, TypeError) as error:
+                    display(args, "  - Error obtaining file size in {}: {}".format(line, error), 0)
+                try:
+                    name = item[1]
+                except IndexError:
+                    display(args, "  - Error obtaining file name in {}".format(line), 0)
+
+                display(args, "  - {} ({} 1k blocks)".format(name, size), 0)
+                required_1k_blocks += size
+                core_files[-1].append(name)
+
+    return required_1k_blocks, core_files
+
+
+def get_available_blocks(args):
+    """Get the local number of available 1k blocks where the core files will be copied.
+
+    Args:
+        args (argparse.Namespace): command line arguments for this program
+
+    Raises:
+        RuntimeError: if there was an error getting the number of available 1k blocks
+
+    Returns:
+        int: the number of available 1k blocks where the core files will be copied
+
+    """
+    command = ["df", "-k", "--output=avail,target", args.destination]
+    try:
+        stdout = run_command(command).splitlines()[-1]
+    except (RuntimeError, AttributeError, ValueError) as error:
+        raise RuntimeError(
```

Comment:
(pylint-raise-missing-from) Consider explicitly re-raising using the 'from' keyword

Reply:
Fixed